src/python/pants/util/dirutil.py

```diff
@@ -83,6 +83,21 @@ def safe_mkdir_for(path, clean=False):
   safe_mkdir(os.path.dirname(path), clean=clean)
 
 
+def safe_mkdir_for_all(paths):
+  """Make directories which would contain all of the passed paths.
+
+  This avoids attempting to re-make the same directories, which may be noticeably expensive if many
+  paths mostly fall in the same set of directories.
+  :type paths list<string>
```

Comment:
I don't think I've seen this param annotation style before?

Reply:
```
pants$ grep -rI ':type ' src/python | wc -l
     108
```

Happy to take alternative suggestions though; I've never done real pydoc before, and am just grasping at what I find it the codebase :)

Reply:
Well this is more compact:
`:param list paths: The paths to create dirs for.`

More verbose:

`:param paths: The paths to create dirs for.`
`:type paths: list of str`

And if you don't want to document the meaning of the param, you can omit the first line I guess. 

But importantly, you need that second `:`, and the way to say "list of str" is `list of str`...

Hopefully once we're all Python3 all the time, we can switch to using https://www.python.org/dev/peps/pep-0484/ instead.

Reply:
Done, maybe? :)